File: Protogen/loxim_protocol/cpp/protocol/packages/W_c_authorizedPackage.cpp
#include <string.h>
#include <stdlib.h>
#include "W_c_authorizedPackage.h"
#include "ptools/PackageBufferWriter.h"
#include "ptools/PackageBufferReader.h"

using namespace protocol;

W_c_authorizedPackage::W_c_authorizedPackage()
{
}

W_c_authorizedPackage::~W_c_authorizedPackage()
{
}

void W_c_authorizedPackage::serializeW(PackageBufferWriter *writer)
{
	Package::serializeW(writer);
}

bool W_c_authorizedPackage::deserializeR(PackageBufferReader *reader)
{
	if (!Package::deserializeR(reader)) return false;
	return true;
}

bool W_c_authorizedPackage::equals(Package *_package)
{
	if (!_package) return false;
	if (_package->getPackageType()!=this->getPackageType()) return false;
	if (!Package::equals(_package)) return false;
	W_c_authorizedPackage* p=(W_c_authorizedPackage*)_package;
	return true;
}

File: Protogen/loxim_protocol/java/src/main/java/pl/edu/mimuw/loxim/protocol/packages/W_c_authorizedPackage.java
package pl.edu.mimuw.loxim.protocol.packages;

import java.util.EnumSet;
import pl.edu.mimuw.loxim.protogen.lang.java.template.ptools.Package;
import pl.edu.mimuw.loxim.protogen.lang.java.template.exception.ProtocolException;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageInputStreamReader;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageOutputStreamWriter;


import pl.edu.mimuw.loxim.protocol.enums.*;


public  class W_c_authorizedPackage extends Package {
		public final static short ID=(short)14;
	
    	
	public W_c_authorizedPackage()
	{
	}

		
		@Override
	public long getPackageType() {
		return (long)ID;
	}
		
	@Override
	protected void deserializeW(PackageInputStreamReader reader)
			throws ProtocolException {
		super.deserializeW(reader);
		
			}
	
	@Override
	protected void serializeW(PackageOutputStreamWriter writer)
			throws ProtocolException {
		super.serializeW(writer);
			}
	
	@Override
	public boolean equals(Object obj) {
		if(!super.equals(obj))
			return false;
		W_c_authorizedPackage p=(W_c_authorizedPackage)obj;
	    		return true;
	}
	
//================= GETTERS AND SETTERS ===================

    
	 
}


